src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/PropertyDescriptor.cs

```diff
@@ -342,7 +343,15 @@ protected override object GetInvocationTarget(Type type, object instance)
         /// <summary>
         /// Gets a type using its name.
         /// </summary>
-        protected Type GetTypeFromName(string typeName)
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "typeName is annotated with DynamicallyAccessedMembers, which will preserve the type.")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2073:ReturnValueNotSatisfied",
+            Justification = "typeName is annotated with PublicConstructors, which preserves the ctors.")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2072:UnrecognizedReflectionPattern",
```

Comment:
If the type name is "MyType, MyAsm" and `ComponentType` is from "OtherAsm" the below will effectively try to get "MyType, OtherAsm". But if that type exists, linker will NOT preserve it.

The comment below seems to hint that this is solely because of assembly version mismatches... if that's the case, then I agree this is OK. (It would assume the ComponentType is from the same assembly simple name).

Reply:
Good catch! I changed this to be `RequiresUnreferencedCode` because of this.